src/sagemaker_containers/_intermediate_output.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+from concurrent.futures import ThreadPoolExecutor
+from inotify_simple import INotify, flags
```

Comment:
Import modules, not classes and functions https://github.com/google/styleguide/blob/gh-pages/pyguide.md#224-decision

That can give you collision and naming problems in the future. Mocking is messy, for example.